06_puppy/runnerdave/map_store.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import "fmt"
+
+// MapStore represents a simple map storage for the Puppy store
+type MapStore struct {
+	puppies map[uint16]Puppy
+}
+
+// NewMapStore creates a new in-memory store with map intialised
+func NewMapStore() *MapStore {
+	var m MapStore
+	m.puppies = make(map[uint16]Puppy)
+	return &m
+}
+
+// CreatePuppy saves new puppy if not in store, if it is already returns error
+func (m *MapStore) CreatePuppy(p *Puppy) error {
+	if _, ok := m.puppies[p.ID]; ok {
```

Comment:
Some think you should provision ID here, up to you to decide in that regard.

Reply:
I just want to keep it simple for now

Reply:
Absolute fine from my perspective.